.github/workflows/ci.yml

```diff
@@ -0,0 +1,27 @@
+name: Build and package TiddlyDesktop
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build-and-package:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "💾 Checking out repository code..."
+        uses: actions/checkout@v4
+      - name: "🛝 Downloading nw.js..."
+        run: ./download-nwjs.sh
+      - name: "🛠️ Building TiddlyDesktop..."
+        run: ./bld.sh
+      - name: "📦 Packaging TiddlyDesktop..."
+        run: ./package.sh
+      - name: "🚀 Uploading packages..."
+        uses: actions/upload-artifact@v4
```

Comment:
No doubt just my unfamiliarity with GitHub Actions, but where do the uploaded files end up?

Reply:
Maybe this only needs to run when a release tag is applied? We'd need suport for prereleases as well as proper releases.

Reply:
These files end up [stored on GitHub's servers](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts) and kept for 90 days (by default) before being automatically deleted. (You can change the retention settings, setting them as low as 1 day if you want). Free accounts get 500 Mb of storage per repository for build artifacts, and paid accounts can, of course, pay more to increase that limit if they want to. After each workflow run, when you look at the workflow output, there's a "Build artifacts" section near the bottom where you can download the files (see https://docs.github.com/en/actions/managing-workflow-runs/downloading-workflow-artifacts for details).

You can also use the `actions/download-artifacts` action to download artifacts in a subsequent workflow run. This is useful, for example, in a build-test-deploy setup where each of those is set up as a different job. (Each job in GitHub Actions runs in a separate, clean run environment that does not contain files from previous jobs, so if you want to share data between jobs, you either use job outputs for short text strings, or artifacts for larger files). The build step uploads the artifacts, then the test step downloads them and runs unit tests or integration tests on them, and if the test job succeeded then the deploy step downloads the artifacts again and creates a release with a new version number.

Reply:
Note that these are *not* permanently stored. With your comment about this running when a release tag is applied, I think you were thinking that this uploads the artifacts to a new release. There's a separate [actions/gh-release](https://github.com/marketplace/actions/gh-release) action for that, or you can use [the manual process](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release). If you use the `actions/gh-release` action, then it would run as a separate job that would do the following:

- Run `actions/download-artifacts` to download the .zip files created by an earlier step
- Create release notes automatically from a RELEASE_NOTES.md or [CHANGELOG.md](https://keepachangelog.com/en/1.1.0/) file
    - Alternately, you can leave the release notes blank and edit them later; if so, you'll want a draft release (see below)
- Run `actions/gh-release` to upload the artifact files to the release
    - You might want to set `draft: true` and verify that the right files got uploaded, then edit the release notes and then promote the draft to a "real" release; while it's a draft, only repo maintainers can see it and the general public can't see it yet.